File: Inventaire/Repository/Models/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Threading.Tasks;

namespace Repository.Models
{
    public class Product
    {
        [DatabaseGenerated(DatabaseGeneratedOption.Identity)]
        [Key]
        public Guid ProductID { get; set; }

        [Required]
        public int CategoryID { get; set; }

        [ForeignKey("CategoryID")]
        public Category Category { get; set; }

        [Required]
        [MaxLength(100)]
        public string Name { get; set; }

        [Required]
        [MaxLength(255)]
        public string Description { get; set; }

        [Required]
        public float Weight { get; set; }
    }
}

File: Inventaire/web-app/types/product.ts
import { Category } from './category';

export interface Product {
  productID: string;
  category: Category;
  name: string;
  description: string;
  weight: number;
}

export interface ProductName {
  productID: string;
  name: string;
}
export interface ProductToAPI {
  productID: string;
  categoryID: number;
  name: string;
  description: string;
  weight: number;
}

export interface addProductToBD {
  categoryID: number;
  name: string;
  description: string;
  weight: number;
}
